scripts/automatic_xml_preprocess.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
```

Comment:
This script will be linked to the build/bin directory when the user runs make geosx_xml_tools.  If you substitute geosx for geosx_preprocessed, then this script will intercept the preprocessor arguments, compile the final input xml file, and then forward to GEOSX.

Right now, it's not MPI-aware, so each process will do this in parallel.  I'm not sure if this will be an issue, or how to best avoid this yet.  We could require mpi4py as a pre-requisite, which is already going to be in the pygeosx distribution, or we could do some simple checks to look if there is a recently written output file.

Reply:
> Right now, it's not MPI-aware, so each process will do this in parallel. I'm not sure if this will be an issue, or how to best avoid this yet.

This surely can be an issue when running on a lot of cores: the filesystem could be clogged by a lot of request in the same time at the same spot (I've experienced it). Now it may be ok on fs with good caching systems I imagine.